File: Open Source Society University Courses/001 HarvardX CS50/CS50 Week 1/pset1/greedy.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

//Function to print the total number of coins needed.
void CoinsNeeded(int);

int main(void){
    
    //Variables to hold the change the user inputs (floatChange) and
    //another to hold the change after being converted to an int (change).
    int change;
    float floatChange;
    
    //Asks user for a float greater than to 0.
    do{
    
    printf("How much change is there?: ");
    floatChange = get_float();
    
    }while (floatChange < 0);
    
    //Converts the float the user input into an int accurately.
    change = roundf(floatChange * 100);

    CoinsNeeded(change);
    
}

void CoinsNeeded(change){
    
    //Counter to keep track of the number of coins needed.
    int coinCount = 0;
    
    //Checks for quarters, then dimes, nickels, and pennies.
    while (change - 25 >= 0){
        
        change -= 25;
        coinCount++;
        
    }
    
    while (change - 10 >= 0){
        
        change -= 10;
        coinCount++;
        
    }
    
    while (change - 5 >= 0){
        
        change -= 5;
        coinCount++;
        
    }
    
    while (change - 1 >= 0){

        change -= 1;
        coinCount++;
        
    }
    
    //Prints out the minimum number of coins needed.
    printf("%d\n", coinCount);
    
}

File: Open Source Society University Courses/001 HarvardX CS50/CS50 Week 8/pset 6/greedy/greedy.py
import cs50

# get from user amount owed
amountOwed = -1
while amountOwed < 0:
    print("How much is owed?: ", end="")
    amountOwed = cs50.get_float()

# initialize coins to keep track of the total number of coins needed
coins = 0

# while the amount owed doesn't go below 0, keep using coins to reach
# the minimum amount needed to make change for the amount
while amountOwed != 0.0:
    
    if amountOwed - .25 >= 0:
        coins += 1
        amountOwed -= .25
        
    elif amountOwed - .10 >= 0:
        coins += 1
        amountOwed -= .10
    
    elif amountOwed - .05 >= 0:
        coins += 1
        amountOwed -= .05
        
    elif amountOwed - .01 >= 0:
        coins += 1
        amountOwed -= .01
        
# print the number of coins used
print(coins)
        
    

